instagram/src/components/CommentSection/CommentContainer.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import './CommentContainer.css';
+import PropTypes from 'prop-types';
+
+class CommentContainer extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            commentData: this.props.postData.comments,
+            commentValue: ''
+        }
+
+        this.newCommentSubmit = this.newCommentSubmit.bind(this);
+        this.newCommentChange = this.newCommentChange.bind(this);
+        this.checkLiked = this.checkLiked.bind(this);
+    }
+
+    newCommentSubmit(event) {
+        let newComment = [...this.state.commentData]
+        let username = localStorage.getItem('username');
+
+        event.preventDefault();
+
+        newComment.push({id: (Date.now()).toString(), username: username, text: this.state.commentValue});
+        this.setState({commentData: newComment})
+    }
+    
+    newCommentChange(event) {
+        this.setState({commentValue: event.target.value});
+    }
+
+    checkLiked() {
+        if(this.props.postData.liked){
+            return <i onClick={(event) => this.props.likedHandler(event, this.props.postData.id)} className="fas liked fa-heart fa-2x"></i>
+        }
+        else {
+            return <i onClick={(event) => this.props.likedHandler(event, this.props.postData.id)} className="far fa-heart fa-2x"></i>
+        }
+    }
+    
+    render() {
+        return (
+            <div className="commentContainer">
+                {/* Action Icons */}
+                <div className="actionIcons">
+                    <div className="commentLinks">
+
+                        {this.checkLiked()}
+
+                        <i className="far fa-comment fa-2x"></i>
+                    </div>
+
+                    <span><b>{this.props.postData.likes} likes</b></span>
+                </div>
+                
```

Comment:
On line 49 instead of invoking the checkLiked function, put your JSX <i> tag here that you're returning inside the function.

<i onClick={(event) => this.likedHandler(event, this.props.postData.id)} className=`${ternary will happen here}` />

This is pretty much the same as how you had it, the main change is, refactoring your className. We can perform ternary now inside of the className to return either of the font awesome classes (you're just returning a string essentially).

<i onClick={(event) => this.likedHandler(event, this.props.postData.id)} className={this.props.postData.liked ? "fas liked fa-heart fa-2x" : "far fa heart"} />

Now, our className string can swap back and forth every render! It's pretty cool, because now you're only changing the className string, instead of returning an entire JSX tag.